e2e/cypress/integration/user.test.js

```diff
@@ -9,26 +11,56 @@ describe('user', () => {
       method: 'POST',
     });
 
-    cy
-      .get('[data-test-id=email]')
+    cy.get('[data-test-id=email]')
       .type(username)
       .get('[data-test-id=password]')
       .type(password)
       .get('[data-test-id=login]')
       .click();
 
-    cy
-      .waitForRoute('/api/accounts/login')
-      .its('status')
-      .should('equal', 200);
+    cy.waitForRoute('/api/accounts/login').its('status').should('equal', 200);
 
     cy.registerRoute('/api/accounts/logout');
 
     cy.get('[data-test-id=logout]').click();
 
-    cy
-      .waitForRoute('/api/accounts/logout')
-      .its('status')
-      .should('equal', 200);
+    cy.waitForRoute('/api/accounts/logout').its('status').should('equal', 200);
+  });
+
+  onlyOn('headless', () => {
+    it('user session timeout', () => {
+      const username = `cataloger@inspirehep.net`;
+      const password = '123456';
+      cy.clock();
+      cy.visit('/user/login/local');
+
+      cy.registerRoute({
+        url: '/api/accounts/login',
+        method: 'POST',
+      });
+
+      cy.get('[data-test-id=email]')
+        .type(username)
+        .get('[data-test-id=password]')
+        .type(password)
+        .get('[data-test-id=login]')
+        .click();
+
+      cy.waitForRoute('/api/accounts/login').its('status').should('equal', 200);
+
+      cy.clearCookies();
+      cy.request({
+        url: '/api/accounts/me',
+        failOnStatusCode: false,
+      })
+        .its('status')
+        .should('equal', 401);
+
+      cy.window().trigger('mouseover', 'topRight');
+      cy.tick(1800000);
+      cy.clock().invoke('restore');
+      cy.wait(500);
```

Comment:
left overs?

Reply:
Actually, there is only one notification. Since it is sticky, it was staying when scroll down the page. So cypress screenshot took it multiple times.

Reply:
Yes, the notification appears after 1800000 millis (30 mins) of inactivity. `tick` moves the time so we don't need to wait for 30 mins.